drivers/power/ltc4162l/ltc4162l.h

```diff
@@ -0,0 +1,266 @@
+#define LTC4162L_EN_CHARGER_STATE_ALERTS_REG        0x0E
+#define LTC4162L_EN_CHARGE_STATUS_ALERTS_REG        0x0F
+#define LTC4162L_CONFIG_BITS_REG                    0x14
+#define LTC4162L_IIN_LIMIT_TARGET                   0x15
+#define LTC4162L_ARM_SHIP_MODE                      0x19
+#define LTC4162L_CHARGE_CURRENT_SETTING             0X1A
+#define LTC4162L_VCHARGE_SETTING                    0X1B
+#define LTC4162L_C_OVER_X_THRESHOLD                 0x1C
+#define LTC4162L_MAX_CV_TIME                        0X1D
+#define LTC4162L_MAX_CHARGE_TIME                    0X1E
+#define LTC4162L_CHARGER_CONFIG                     0x29
+#define LTC4162L_CHARGER_STATE                      0x34
+#define LTC4162L_CHARGE_STATUS                      0x35
+#define LTC4162L_LIMIT_ALERTS_REG                   0x36
+#define LTC4162L_CHARGER_STATE_ALERTS_REG           0x37
+#define LTC4162L_CHARGE_STATUS_ALERTS_REG           0x38
+#define LTC4162L_SYSTEM_STATUS_REG                  0x39
+#define LTC4162L_VBAT                               0x3A
+#define LTC4162L_VIN                                0x3B
+#define LTC4162L_VOUT                               0x3C
+#define LTC4162L_IBAT                               0x3D
+#define LTC4162L_IIN                                0x3E
+#define LTC4162L_DIE_TEMPERATURE                    0x3F
+#define LTC4162L_THERMISTOR_VOLTAGE                 0x40
+#define LTC4162L_BSR                                0x41
+#define LTC4162L_JEITA_REGION                       0x42
+#define LTC4162L_CHEM_CELLS_REG                     0x43
+#define LTC4162L_ICHARGE_DAC                        0x44
+#define LTC4162L_VCHARGE_DAC                        0x45
+#define LTC4162L_IIN_LIMIT_DAC                      0x46
+#define LTC4162L_VBAT_FILT                          0x47
+#define LTC4162L_INPUT_UNDERVOLTAGE_DAC             0x4B
+
+#define LTC4162L_CELL_COUNT_MASK                    NO_OS_GENMASK(3, 0)
+#define LTC4162L_CHEM_TYPE_MASK                     NO_OS_GENMASK(11, 8)
+#define LTC4162L_CHEM_MASK                          NO_OS_GENMASK(11, 8)
+#define LTC4162L_ONLINE_STATUS_BIT                  NO_OS_BIT(2)
+#define LTC4162L_EN_C_OVER_X_TERM_BIT               NO_OS_BIT(2)
+#define LTC4162L_IIN_LIMIT_DAC_MASK                 NO_OS_GENMASK(6, 0)
+#define LTC4162L_CHARGER_CONFIG_BIT                 NO_OS_BIT(2)
+#define LTC4162L_ICHARGE_DAC_MASK                   NO_OS_GENMASK(4, 0)
+#define LTC4162L_VIN_GT_VBAT_BIT                    NO_OS_BIT(2)
+#define LTC4162L_VCHARGE_SETTING_MASK               NO_OS_GENMASK(4, 0)
+#define LTC4162L_RUN_BSR_BIT                        NO_OS_BIT(4)
+#define LTC4162L_FORCE_TELEMETRY_BIT                NO_OS_BIT(2)
+
+#define LTC4162L_ARM_SHIP_MODE_MAGIC		21325
+#define LTC4162L_MULTIPLIER			14660
+#define LTC4015_MULTIPLIER			146487
+#define LTC4162L_VCHARGE_RANGE			38125
+#define LTC4015_VCHARGE_RANGE			34125
+
+
+#define LTC4162L_MULT_FRAC(x, mult, div) ((x * mult) / div)
```

Comment:
i don't see the need of this macro, and besides it has an error - if `div` is an expression like `2 * 2` or `1 <<  3` it will result in incorrect computation, i suggest you remove it completely (or fix it if you don't want to remove it)

Reply:
tbh, all of its uses in the file are okay, they are not expressions, they are either constants or variables, but someone's gonna come next and add a feature to the driver, use the macro with an expression and will waste effort debugging this

Reply:
Nice catch. Looking at the usage of this, I think it is shorter to just write the expression directly than use a macro for this. There's also not that many use.